### PR TITLE
docs: update user-flow.md to reflect current release

### LINKED ISSUE #14603: User Flow docs are not up-to-date?
**Summary**

The import path for startFlow on the docs for [**User Flows in Lighthouse**](https://github.com/GoogleChrome/lighthouse/blob/03674e9b100f769367e4f174ba7ca29a5b081912/docs/user-flows.md) are not the same as the current release or the version repo is on now. 

The import path specifies on in the documentation is:

```javascript
import lighthouse from 'lighthouse/core/api.js';
```

The import path for the latest release [**v9.6.8**](https://github.com/GoogleChrome/lighthouse/releases/tag/v9.6.8) is:

 ```javascript
 import lighthouse from 'lighthouse/lighthouse-core/fraggle-rock/api.js';
 ```
 
And the import path on the current repo is:
 
  ```javascript
  import lighthouse from 'lighthouse/core/index.js';
  ```
  
 I can also see the docs have a cool feature `startNavigation` and `endNavigation`.
 
Which is the correct path?

1. Path on current Documentation
2. Path on latest release
3. Path on curent repo main
   
Can I open a pr and update the docs? :)

### PR DESCRIPTION
**Pull request by ChristopherPHolder**
**Summary**

Updating the docs for User Flow to reflect the current release

- Change flow import path from `import lighthouse from 'lighthouse/core/api.js';` to `import {startFlow} from 'lighthouse/lighthouse-core/fraggle-rock/api.js';`

- remove `startNavigation`/`endNavigation`

- change interactive navigation with `startNavigation`/`endNavigation` to a callback navigation

- change `beginTimepsan` to `startTimespan`

**Related Issues/PRs**

Closes #14603

### CODE REVIEW

File: docs/user-flows.md [hunk 1]
```diff
@@ -90,15 +83,15 @@ In DevTools, select "Timespan" as the mode and click _Start timespan_. Record wh
...
 
 (async function() {
   const browser = await puppeteer.launch();
   const page = await browser.newPage();
   await page.goto('https://secret.login');
-  const flow = await lighthouse.startFlow(page);
+  const flow = await startFlow(page);
 
-  await flow.beginTimespan();
+  await flow.startTimespan();
```
Comment (adamraine) [inline docs/user-flows.md:94]: Nice catch!

File: docs/user-flows.md [hunk 2]
```diff
@@ -182,15 +175,13 @@ async function search(page) {
...
+  const flow = await startFlow(page);
 
   // Phase 1 - Navigate to the landing page.
   await flow.navigate('https://www.bestbuy.com');
 
   // Phase 2 - Interact with the page and submit the search form.
-  await flow.startTimespan();
-  await search(page);
-  await flow.endTimespan();
+  await flow.navigate(async () => await search(page));
```
Comment (adamraine) [inline docs/user-flows.md:184]: This is replacing a timespan with a navigation, can you revert? `startTimespan`/`endTimespan` still work.

File: docs/user-flows.md [hunk 3]
```diff
@@ -75,8 +70,6 @@ Instead of providing a URL to navigate to, you can provide a callback function o
 
 This callback function _must_ perform an action that will trigger a navigation. Any interactions completed before the callback promise resolves will be captured by the navigation.
 
-The `startNavigation`/`endNavigation` functions _must_ surround an action that triggers a navigation. Any interactions completed after `startNavigation` is invoked and before `endNavigation` is invoked will be captured by the navigation.
```
Comment (adamraine) [inline docs/user-flows.md:78]: This is good, but can you remove the reference to `startNavigation` / `endNavigation` on L72

**Review by adamraine — CHANGES_REQUESTED** [review]
Thanks for the PR! This is what I was imagining on `main`.

**Review by ChristopherPHolder — COMMENTED** [review]
Should we also remove the reference to `(Lighthouse v10)` in L3?

[Historically, Lighthouse has analyzed the cold pageload of a page. Starting in 2022 (Lighthouse v10), it can analyze and report on the entire page lifecycle via "user flows".](https://github.com/GoogleChrome/lighthouse/blob/03674e9b100f769367e4f174ba7ca29a5b081912/docs/user-flows.md?plain=1#L3)

File: docs/user-flows.md [hunk 4]
```diff
@@ -181,7 +181,9 @@ async function search(page) {
   await flow.navigate('https://www.bestbuy.com');
 
   // Phase 2 - Interact with the page and submit the search form.
-  await flow.navigate(async () => await search(page));
+  await flow.startTimespan();
+  await search(page);
+   await flow.endTimespan();
```
Comment (adamraine) [inline docs/user-flows.md:186]:
```suggestion
  await flow.endTimespan();
```

**Review by adamraine — COMMENTED** [review]
> Should we also remove the reference to (Lighthouse v10) in L3?

Yeah sounds good.

**Review by adamraine — APPROVED** [review]

**Pull request merged**